static/app/views/settings/projectTags/index.tsx

```diff
@@ -7,140 +7,140 @@ import {Button} from 'sentry/components/button';
+import useProjects from 'sentry/utils/useProjects';
 import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
 import TextBlock from 'sentry/views/settings/components/text/textBlock';
 import PermissionAlert from 'sentry/views/settings/project/permissionAlert';
 
-type Props = RouteComponentProps<{projectId: string}, {}> & {
-  organization: Organization;
-  project: Project;
-} & DeprecatedAsyncView['props'];
-
-type State = {
-  tags: Array<TagWithTopValues>;
-} & DeprecatedAsyncView['state'];
-
-class ProjectTags extends DeprecatedAsyncView<Props, State> {
-  getDefaultState(): State {
-    return {
-      ...super.getDefaultState(),
-      tags: [],
-    };
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    const {organization} = this.props;
-    const {projectId} = this.props.params;
-    return [['tags', `/projects/${organization.slug}/${projectId}/tags/`]];
-  }
-
-  getTitle() {
-    const {projectId} = this.props.params;
-    return routeTitleGen(t('Tags'), projectId, false);
-  }
-
-  handleDelete = (key: TagWithTopValues['key'], idx: number) => async () => {
-    const {organization, params} = this.props;
-    const {projectId} = params;
-
-    try {
-      await this.api.requestPromise(
-        `/projects/${organization.slug}/${projectId}/tags/${key}/`,
-        {
-          method: 'DELETE',
-        }
+type Props = RouteComponentProps<{projectId: string}, {}>;
+
+type DeleteTagResponse = unknown;
+type DeleteTagVariables = {key: TagWithTopValues['key']};
+
+function ProjectTags(props: Props) {
+  const organization = useOrganization();
+  const {projects} = useProjects();
+  const projectId = props.params.projectId;
+
+  const project = useMemo(
+    () => projects.find(p => p.id === projectId),
+    [projects, projectId]
+  );
+
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  const {data: tags, isLoading} = useApiQuery<TagWithTopValues[]>(
+    [`/projects/${organization.slug}/${projectId}/tags/`],
+    {staleTime: 0}
+  );
+
+  const {mutate} = useMutation<DeleteTagResponse, RequestError, DeleteTagVariables>({
+    mutationFn: ({key}: DeleteTagVariables) =>
+      api.requestPromise(`/projects/${organization.slug}/${projectId}/tags/${key}/`, {
+        method: 'DELETE',
+      }),
+    onSuccess: (_, {key}) => {
+      setApiQueryData<TagWithTopValues[]>(
+        queryClient,
+        [`/projects/${organization.slug}/${projectId}/tags/`],
+        oldTags => oldTags.filter(tag => tag.key !== key)
       );
+    },
+  });
 
-      const tags = [...this.state.tags];
-      tags.splice(idx, 1);
-      this.setState({tags});
-    } catch (error) {
-      this.setState({error: true, loading: false});
-    }
-  };
-
-  renderBody() {
-    const {project} = this.props;
-    const {tags} = this.state;
-    const isEmpty = !tags || !tags.length;
+  if (isLoading) {
+    return <LoadingIndicator />;
+  }
```

Comment:
Should we do error handling too?

@malwilley I think we really should try and build out a helper for doing the loading and error state stuff 😬